### PR TITLE
feat: add handling for clarity-js reference

### PR DESCRIPTION
**Pull request by friedger**
## Description

References of javascript libraries should be included in the docs using the generated typedoc files.

This PR 
* adds code to handle generated markdown files by typedoc for javascript libraries.
* adds generated md files for clarity-js
* adds a json file containing javascript libraries to be included (`libs.json`)

For details refer to issue #846

## Type of Change
- [ ] New feature
- [ ] Bug fix
- [x] API reference/documentation update
- [ ] Other

## Checklist
- [x] Tag @agraebe

### CODE REVIEW

File: src/pages/references/clarity-js/clarity-js.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+---
+title: '@blockstack/clarity'
```
Comment (agraebe) [inline src/pages/references/clarity-js/clarity-js.md:2]: I'd suggest Clarity JS SDK

File: src/common/data/clarity-js-ref.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import * as fs from 'fs';
+import renderToString from 'next-mdx-remote/render-to-string';
+
+export const convertClarityJSRefToMdx = async () => {
+  const md = fs.readFileSync('./src/_data/clarity-js/globals.md');
```
Comment (agraebe) [inline src/common/data/clarity-js-ref.ts:5]: I'm seeing a `Error: ENOENT: no such file or directory, open './src/_data/clarity-js/globals.md'` - you are probably already aware bc this is a draft PR

File: src/pages/references/clarity-js/clarity-js.md (hunk 1 above)
Comment (friedger) [reply src/pages/references/clarity-js/clarity-js.md:2]: Clarity JS SDK is the whole repository. There is an open issue how to properly name this package https://github.com/blockstack/clarity-js-sdk/issues/47

File: src/common/data/clarity-js-ref.ts (hunk 2 above)
Comment (friedger) [reply src/common/data/clarity-js-ref.ts:5]: @agraebe  I couldn't commit the md files because of lint prettier issues.

Comment (friedger) [reply src/common/data/clarity-js-ref.ts:5]: `lint:prettier:fix` did not help

Comment (friedger) [reply src/common/data/clarity-js-ref.ts]: I have added the files now

**Comment by friedger** [comment]
@agraebe I have generalized the js lib reference. There is no clarity-js specific code anymore.

The current setup to add a new js lib reference is as follows:
1. add library id, library path and package name to `/src/_data/libs.json`
1. see the reference added

Missing pieces:
* add github actions for each js lib to import typedoc files
* fix links in md files (the link contains [lib]/[type] path)

File: src/pages/references/libs/[lib]/index.tsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      return { params: { lib: libKey } };
+    }),
+    fallback: false,
+  };
+};
+
+export const getStaticProps = async ({ params }) => {
+  const libPath = stacksLibs[params.lib].path;
+  const md = fs.readFileSync(`./src/_data/${libPath}/globals.md`);
+  const mdxSource = await renderToString(md.toString(), { components: Components });
```
Comment (friedger) [inline src/pages/references/libs/[lib]/index.tsx:29]: Rendered links contain `[lib]` in the path :hankey:

File: src/pages/references/libs/[lib]/[type]/[file].tsx [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    fallback: false,
+  };
+};
+
+export const getStaticProps = async ({ params }) => {
+  const lib = stacksLibs[params.lib];
+  const type = params.type;
+  const file = params.file;
+  const md = fs.readFileSync(`./src/_data/${lib.path}/${type}/${file}`);
+  const mdxSource = await renderToString(md.toString(), { components: Components });
```
Comment (friedger) [inline src/pages/references/libs/[lib]/[type]/[file].tsx:40]: Rendered links contain `[lib]/[type]` in their path :hankey:

**Comment by agraebe** [comment]
thanks @friedger - I just started looking at this. what libs did you include in this and how could I preview the changes in Gitpod? Is there a URL I could open to check it out?

**Comment by friedger** [comment]
@agraebe try references/lib/clarity-js

**Comment by agraebe** [comment]
what do we think should be the next step here?

**Comment by agraebe** [comment]
closing due to inactivity

**Pull request closed without merging**